frame/offences/benchmarking/src/lib.rs

```diff
@@ -403,13 +396,14 @@ benchmarks! {
 		}
 
 		Offences::<T>::set_deferred_offences(deferred_offences);
-		assert!(!Offences::<T>::deferred_offences().is_empty());
+		assert_eq!(Offences::<T>::deferred_offences().unwrap().len(), d as usize);
+		assert_eq!(System::<T>::event_count(), 0);
 	}: {
-		Offences::<T>::on_initialize(0u32.into());
+		Offences::<T>::on_runtime_upgrade();
 	}
 	verify {
 		// make sure that all deferred offences were reported with Ok status.
-		assert!(Offences::<T>::deferred_offences().is_empty());
+		assert!(Offences::<T>::deferred_offences().is_none());
 		assert_eq!(
 			System::<T>::event_count(), d * (0
 			+ o // offenders slashed
```

Comment:
Here I check if the offences have been submitted, I'm not comparing the Offence content, only quantity of events emitted before and after `on_runtime_upgrade`